src/main/java/hudson/remoting/UserRequest.java

```diff
@@ -90,6 +90,17 @@ public UserRequest(Channel local, Callable<?,EXC> c) throws IOException {
         this.classLoaderProxy = RemoteClassLoader.export(cl, local);
     }
 
+    @Override
+    public void checkIfCanBeExecutedOnChannel(Channel channel) throws IOException {
+        // Default check for all requests
+        super.checkIfCanBeExecutedOnChannel(channel);
+        
+        // We also do not want to run UserRequests when the channel is being closed
+        if (channel.isClosingOrClosed()) {
+            throw new IOException("The request cannot be executed on channel " + channel + ". The channel is closing down or has closed down");
```

Comment:
> from `Request#call()` directly (also a public API)

While it might have been `public` by some accident, clearly it was only intended to be called from `Channel.call`, and any other call site should be considered unsupported and the behavior undefined.